### PR TITLE
[Merged by Bors] - refactor(ModelTheory): Redefine order and graph languages to use inductive definitions instead of `Language.mk₂`

### PR DESCRIPTION
**Pull request by awainverse**
Redefines order and graph languages using the same style of inductive definition as `Language.ring`

---

- [x] depends on: #16323
- [x] depends on: #16338

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by awainverse** [comment]
I'm tempted to push this PR even further and eliminate `Language.mk₂`, because working with these inductive definitions is more pleasant, but `constantsOn` is defined that way, and is a little more complicated.

File: Mathlib/ModelTheory/Graph.lean [hunk 1]
```diff
@@ -106,17 +106,12 @@ theorem structure_simpleGraphOfStructure [S : Language.graph.Structure V] [V ⊨
...
-          change RelMap adj ![xs 0, xs 1] = _
-          refine congr rfl (funext ?_)
-          simp [Fin.forall_fin_two]
-        · exact r.elim
+    match n, r with
+    | 2, .adj =>
+      rw [iff_eq_eq]
+      change RelMap adj ![xs 0, xs 1] = _
+      refine congr rfl (funext ?_)
+      simp [Fin.forall_fin_two]
```
Comment (YaelDillies) [inline Mathlib/ModelTheory/Graph.lean:114]: NIce :)

**Review by YaelDillies — APPROVED** [review]
A clear improvement

maintainer merge

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**